onyo/lib/utils.py

```diff
@@ -0,0 +1,76 @@
+from pathlib import Path
+from typing import Dict, Iterable, Union
+
+from ruamel.yaml import YAML, scanner
+
+from .onyo import Repo, log
+
+
+def is_protected_path(path: Union[Path, str]) -> bool:
+    """
+    Checks whether a path contains protected elements (.anchor, .git, .onyo).
+    Returns True if it contains protected elements. Otherwise False.
+    """
+    full_path = Path(path).resolve()
+
+    # protected paths
+    for p in full_path.parts:
+        if p in ['.anchor', '.git', '.onyo']:
+            return True
+
+    return False
+
+
+def get_assets(repo: Repo) -> set[Path]:
```

Comment:
After seeing this change, I am not fully convinced yet, that `get_assets()` and `get_templates()` should be excluded from the repo. It is not clear to me, that "just git handling" is the line that we should draw between onyo.py and utils.py.
Seeing this moved to here gives me the feeling that I know have to remember if functions do internally do a git-call or are requested or cached in some other way then for example `Repo.get_dirs()`.

I personally would prefer this to be split like the hypothetical asset class, where `Repo` contains all stuff related to handling of repo-information, independently from where and how the information is found.

That said, +1 on the `read_asset()`, `is_protected_path()` and all the `fsck()` related functios in `command_utils.py` :)

Reply:
> Seeing this moved to here gives me the feeling that I know have to remember if functions do internally do a git-call or are requested or cached in some other way then for example Repo.get_dirs().

To add to conversation in chat: No, you don't need to know. In this state, you'd still call `Repo.get_assets`/`Repo.get_dirs` as before. This is all about being able to reshape things w/o interfering in the usage of `Repo` yet.